src/textual/layouts/horizontal.py

```diff
@@ -26,7 +26,9 @@ def arrange(
         add_placement = placements.append
 
         child_styles = [child.styles for child in children]
-        box_margins: list[Spacing] = [styles.margin for styles in child_styles]
+        box_margins: list[Spacing] = [
+            styles.margin for styles in child_styles if styles.overlay != "screen"
+        ]
```

Comment:
I see you changed this because you probably thought of an edge case that wasn't being covered; wouldn't this warrant a test?

Reply:
It probably should. Going to leave it for now, because overlay isn't yet documented.